CMakeLists.txt

```diff
@@ -116,6 +119,24 @@ if(NOT HAS_ACCEPTABLE_FORTIFY)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${_FORTIFY_SOURCE_PREFIX}-U_FORTIFY_SOURCE ${_FORTIFY_SOURCE_PREFIX}-D_FORTIFY_SOURCE=1")
 endif()
 
+# Remove --sort-common from linker flags, as this seems to cause bugs (see #2641, #3374).
+# TODO: Figure out the root cause.
```

Comment:
FWIW `--sort-common` appears to be used with Vim by several people, based on a vim_dev search:

https://groups.google.com/forum/#!searchin/vim_dev/sort$20common

So this may be something we introduced.